crates/sui-indexer-alt-jsonrpc/src/api/coin.rs

```diff
@@ -123,6 +133,39 @@ impl CoinApiServer for CoinServer {
             .await
             .with_internal_context(|| format!("Failed to get coins for {owner}"))?)
     }
+
+    async fn get_all_balances(&self, owner: SuiAddress) -> RpcResult<Vec<Balance>> {
+        let coin_ids = self.get_coin_id_page(owner, None, None, None).await?;
+        let coin_futures = coin_ids.data.iter().map(|id| self.coin_response(*id));
```

Comment:
Fetching the full coin response seems a bit redundant, instead of fetching and processing the `StoredObject`-s, but I can live with it given that `coin_response` does minimal extra work. If there were a way to factor out the common part into its own function, that would be better (see how I've factored `live_object`, `past_object`, and `object_data` for the object queries).